### PR TITLE
feat(date-picker): Add OnPush change detection strategy

### PR DESCRIPTION
**Pull request by ert78gb**
close: #168

### CODE REVIEW

**Comment by vlio20** [comment]
Thanks for the PR.
Could you check out why your changed didn't pass in Travis?

**Comment by ert78gb** [comment]
of course I will check tomorrow or after tomorrow.

**Comment by ert78gb** [comment]
All green :)

**Comment by vlio20** [comment]
This OnPush feature is cool, but it seems like it adds a lot of overhead while developing, don't you agree? what is the benefit - does it really significantly improves the performance?

**Comment by ert78gb** [comment]
If you keep a few rules then there is no overhead. Example:
- The external component is the main controller
- The main controller communicate with the children only with immutable @Input properties.

I don't how many the performance improvement because all of my app use OnPush. If one component is not support OnPush in the component tree than the rendering will not correct. So it is need for me and I think to others too.

It is a cool component.

**Comment by ert78gb** [comment]
Do you plan to accept the PR today?

File: src/app/date-picker/date-picker.component.ts [hunk 1]
```diff
@@ -326,7 +340,7 @@ export class DatePickerComponent implements OnChanges,
   showCalendars() {
     this.hideStateHelper = true;
     this.areCalendarsShown = true;
-
+    this.cd.markForCheck();
     if (this.timeSelectRef) {
```
Comment (vlio20) [inline src/app/date-picker/date-picker.component.ts]: please add line break

**Comment by ert78gb** [comment]
what do you mean by that exactly?

File: src/app/date-picker/date-picker.component.ts (hunk 1 above)
Comment (ert78gb) [reply src/app/date-picker/date-picker.component.ts]: I added before and after the new line

**Comment by vlio20** [comment]
Add to the readme file of this project a note that specifies that this component is supporting OnPush methodology.

**Comment by ert78gb** [comment]
ok I did.

**Comment by ert78gb** [comment]
thx

**Comment by vlio20** [comment]
@ert78gb, there were another merge earlier and it seems that your changes are breaking Travis.

**Comment by ert78gb** [comment]
I will check it soon, but until could you restart the travis build because it is a timeout maybe just a timeout

**Comment by vlio20** [comment]
just did

**Comment by ert78gb** [comment]
So it was a 'simple' timeout :)

**Comment by vlio20** [comment]
thanks @ert78gb

**Comment by ert78gb** [comment]
your welcome

**Pull request merged**